libs/configuration-variables-resolver/tests/VariableResolverTest.php

```diff
@@ -372,16 +372,10 @@ public function testResolveVariablesNoVariables(): void
             'parameters' => ['some_parameter' => 'foo is {{ foo }}'],
         ];
         $variableResolver = $this->getVariableResolver();
-        $newConfiguration = $variableResolver->resolveVariables($configuration, '123', []);
-        self::assertEquals(
-            [
-                'parameters' => [
-                    'some_parameter' => 'foo is {{ foo }}',
-                ],
-            ],
-            $newConfiguration
-        );
-        self::assertFalse($this->testLogger->hasInfoThatContains('Replacing variables using default values with ID:'));
+
+        $this->expectException(UserException::class);
+        $this->expectExceptionMessage('Missing values for placeholders: foo');
+        $variableResolver->resolveVariables($configuration, '123', []);
```

Comment:
Tohle je BC break, protoze pokud konfigurace obsahuje mustache pattern, zacne nove failovat kvuli neexistujici variable. Nenapada me ale jak to udelat lip, bez nejakeho feature flagu nebo obskurnich magickeho checkovani.

Driv, kdyz konfigurace nemela `variables_id`, tak se vubec neresil replacement promennych. Tim padem zustaly mustache placeholdery v konfiguraci a jelo se dal. Jelikoz nove budeme nacitat i z Vaultu, `variables_id` nastavene byt nemusi - skipne se nacitani z konfigurace, ale porad se nacitaji promenne z Vaultu.

Reply:
Udelal bych to zatim na tu SOX featuru. 

Muze se totiz stat, ze pro nejakou komponentu chces pouzit zrovna stejny pattern jako my pro nase Variables a timhle to uplne rozbijem/znemoznime.

Do budoucna by to pak chtelo mozna nejaky flag pro komponentu, ze nemuze pouzivat nase variables, ale to bych zatim takhle neresil.

Reply:
Ok, takze nakonec se to cely prepsalo :slightly_smiling_face: a Vault ma vlastni resolver.